### PR TITLE
fix: Left Sidebar Not moving down 

### LINKED ISSUE #6534: Public Event Page: Left Sidebar not longer moving down on scroll down
One of the recent changes changed the behavior of the left sidebar. The left sidebar used to move down when the user scrolls down the page. This behavior disappeared. Please re-instate this sidebar behavior.

Compare https://eventyay.com/e/fa96ae2c/

Screenshot from 2021-01-29 20-50-07

### PR DESCRIPTION
**Pull request by dhruvin1831**
**Fixes #6534** 

#### Short description of what this resolves:
It fixes the sidebar scrolling issue.

#### Changes proposed in this pull request:
Used Uisticky  component of semantic ui ember for sticky.
-
-
-

https://user-images.githubusercontent.com/70140023/107090476-445e2a80-6826-11eb-99e5-593fa0be2956.mp4

Screenshot 2021-02-06 022911

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/styles/pages/public.scss [hunk 1]
```diff
@@ -12,3 +16,9 @@
   display: flex;
   flex-wrap: wrap;
 }
+
+@media screen and (min-width: 768px) and (max-width: 992px) {
+  .public-container {
+    padding: 0 .8rem;
+  }
+}
```
Comment (iamareebjamal) [inline app/styles/pages/public.scss:24]: Not correct

Comment (dhruvin1831) [reply app/styles/pages/public.scss:24]: Sir, may you please clarify that the bug is not fixed or the way of doing is the issue?

Comment (iamareebjamal) [reply app/styles/pages/public.scss:24]: Way of doing

File: app/styles/pages/public.scss [hunk 2]
```diff
@@ -12,3 +12,13 @@
...
   flex-wrap: wrap;
 }
+
+.rightside-menu {
+  overflow-x: hidden;
+}
+
+@media screen and (min-width: 769px) {
+  .public-container {
+    max-width: 100vw;
```
Comment (dhruvin1831) [inline app/styles/pages/public.scss:22]: Max width was the issue with the container which was crossing the width of the device.

**Comment by dhruvin1831** [comment]
@iamareebjamal Sir, Please review the last commit.

**Comment by iamareebjamal** [comment]
What's the need of media query

**Comment by dhruvin1831** [comment]
> What's the need of media query

@iamareebjamal Because of how the containers are arranged in the mobile view (width < 769px). footer-pusher's overflow-x property takes care of scroll bar in mobile view on any page, but adding max-width to mobile view(width < 769px) is unnecessary as well will create collapsing margins because of negative and positive margins of adjacent containers in mobile view (particularly in the public event page due to negative margins of the container). To prevent that I have used a media query.

**Comment by dhruvin1831** [comment]
@iamareebjamal Sir, please review the commit and let me know if further actions are required.

**Comment by dhruvin1831** [comment]
@iamareebjamal Sir, please review the latest commit and let me know if any further actions are required.

File: app/templates/public.hbs [hunk 3]
```diff
@@ -33,9 +33,11 @@
           {{/if}}
         {{else}}
           {{#if this.displaySideMenu}}
+          <UiSticky>
```
Comment (dhruvin1831) [inline app/templates/public.hbs:36]: ref: [https://semantic-ui.com/modules/sticky.html#/usage](url)

File: app/templates/public.hbs [hunk 4]
```diff
@@ -33,9 +33,11 @@
           {{/if}}
         {{else}}
           {{#if this.displaySideMenu}}
+          <UiSticky @offset={{10}}>
```
Comment (dhruvin1831) [inline app/templates/public.hbs:36]: Offset setting was added to uisticky (using UI semantic ember).

File: app/templates/public.hbs [hunk 5]
```diff
@@ -33,9 +33,11 @@
           {{/if}}
         {{else}}
           {{#if this.displaySideMenu}}
+          <UiSticky @offset={{10}} @context={{'#public-event-content'}} @observeChanges={{true}}>
```
Comment (dhruvin1831) [inline app/templates/public.hbs:36]: Used Semantic UI ember sticky component. Added context to stick to.

**Comment by dhruvin1831** [comment]
@iamareebjamal I have fixed the sticky behavior. Please review this and let me know if any further actions are required.

File: app/templates/public.hbs (hunk 5 above)
Comment (dhruvin1831) [reply app/templates/public.hbs:36]: https://semantic-org.github.io/Semantic-UI-Ember/#/modules/sticky

**Comment by mariobehling** [comment]
If you keep scrolling on the speakers page the sidebar disappears at one point. Please test https://open-event-frontend-git-fork-dhruvin1831-sidebarfix-eventyay.vercel.app/e/fa96ae2c/speakers

**Comment by mariobehling** [comment]
https://open-event-frontend-git-fork-dhruvin1831-sidebarfix-eventyay.vercel.app/e/fa96ae2c/schedule

**Comment by dhruvin1831** [comment]
> If you keep scrolling on the speakers page the sidebar disappears at one point. Please test https://open-event-frontend-git-fork-dhruvin1831-sidebarfix-eventyay.vercel.app/e/fa96ae2c/speakers

Yes, this is a problem. it occurs when, The contents of the container to which sidebar is sticky are dynamically loaded in any page on scroll so the lower bound set on the sticky sidebar is crossed as soon as we scroll up to the point it was loaded initially, and sidebar gets fixed to the bottom.

**Comment by dhruvin1831** [comment]
There might be a solution to this using semantic UI. working on this.

**Comment by maze-runnar** [comment]
Screenshot from 2021-03-03 23-37-02

File: app/styles/pages/public.scss [hunk 6]
```diff
@@ -4,8 +4,7 @@
 }
 
 .side-menu {
-  position: sticky;
-  top: 14px;
+  z-index: -1 !important;
```
Comment (dhruvin1831) [inline app/styles/pages/public.scss:7]: @maze-runnar fixed the side-panel overlap issue by adding z-index.

File: app/templates/public.hbs [hunk 7]
```diff
@@ -33,14 +33,16 @@
           {{/if}}
         {{else}}
           {{#if this.displaySideMenu}}
-            <div class="ui fluid side-menu">
+          <UiSticky class="side-menu" @offset={{10}} @context={{'#event-detail'}} @observeChanges={{true}}>
```
Comment (dhruvin1831) [inline app/templates/public.hbs:36]: changed the context to the column which was loading dynamically to solve scrolling issue.

File: app/styles/pages/public.scss [hunk 8]
```diff
@@ -4,8 +4,7 @@
 }
 
 .side-menu {
-  position: sticky;
-  top: 14px;
+  z-index: 0 !important;
```
Comment (dhruvin1831) [inline app/styles/pages/public.scss:7]: @maze-runnar fixed the side-panel overlap issue by adding z-index.

File: app/templates/public.hbs (hunk 7 above)
Comment (dhruvin1831) [inline app/templates/public.hbs:36]: changed the context to the column which was loading dynamically to solve scrolling issue.

Comment (dhruvin1831) [reply app/templates/public.hbs:36]: @mariobehling I have tested this with the previous issue raised by you and it works fine now. Please let me know if any further actions are required.

**Comment by dhruvin1831** [comment]
@iamareebjamal Sir, please review the changes. I have tested the issues and it's working fine now.

File: app/templates/public.hbs [hunk 9]
```diff
@@ -33,14 +33,16 @@
...
+            <div class="ui fluid">
               <Public::SideMenuOuter @event={{this.model}} />
             </div>
+          </UiSticky> 
           {{/if}}
         {{/if}}
       </div>
       {{#unless this.device.isMobile}}
-        <div class="ten wide column">
+        <div class="ten wide column" id="event-detail">
```
Comment (maze-runnar) [inline app/templates/public.hbs:45]: whi added id?

**Pull request closed without merging**